pkg/network/handlers/work_package_sharing.go

```diff
@@ -67,40 +95,82 @@ func (h *WorkPackageSharer) shareWorkPackageWithOtherGuarantors(ctx context.Cont
 				return
 			}
 
-			err = h.sendWorkPackage(ctx, stream, coreIndex, []work.ImportedSegment{}, bundle)
+			defer func() {
+				if err = stream.Close(); err != nil {
+					fmt.Printf("failed to close stream: %v", err)
+				}
+			}()
+
+			err = h.sendWorkPackage(ctx, stream, coreIndex, segments, bundle)
 			if err != nil {
 				log.Printf("Failed to share WP with peer %v: %v", g, err)
 			}
 
-			// TODO:  Once the receive side (CE-134 response) is implemented we should read the response
+			// Handle CE-134 response from the receiving guarantor
+			msg, err := ReadMessageWithContext(ctx, stream)
+			if err != nil {
+				log.Printf("Failed to read response from peer %v: %v", g, err)
+				return
+			}
+
+			var response struct {
+				WorkReportHash crypto.Hash
+				Signature      []byte
+			}
+			if err := jam.Unmarshal(msg.Content, &response); err != nil {
+				log.Printf("Failed to decode CE-134 response from peer %v: %v", g.ValidatorIndex, err)
+				return
+			}
+
+			log.Printf("Received work-report hash and signature from peer %v:\n- Hash: %x\n- Signature: %x",
+				g.ValidatorIndex, response.WorkReportHash, response.Signature)
 		}(g)
 	}
 
+	// start local refinement in parallel
+	wg.Add(1)
```

Comment:
Are we planning on doing more in parallel later, or more work after while this is running? As this stands I don't see how making this parallel is adding any speedup.

Reply:
Sharing the work package with guarantors and running refinement should happen in parallel. In the future we could probably parallelize the refinement (maybe per work item or something like that)

Reply:
Ah my bad I missed the wait group right at the top. Makes sense. 👍